### PR TITLE
Add Relay class for orchestrating hydra runs.

### PR DESCRIPTION
**Pull request by MylesBartlett**

### CODE REVIEW

**Comment by thomkeh** [comment]
I don't see the deletion of hydra.py

**Comment by thomkeh** [comment]
Is the dependency on attrs absolutely necessary?

File: kit/hydra/relay.py [hunk 1]
```diff
@@ -0,0 +1,305 @@
...
+    of the schemas can't be found), with the resulting schemas cached in the config directory.
+
+    Subclasses must implement a 'run' method and will themselves be converted into the
+    primary config classes used to initialise and validate hydra. With the launching of hydra,
+    the subclasses will be instantiated and the run method called on the raw config.
+
+    :example:
+
+    >>>
+    Relay.with_hydra(
```
Comment (thomkeh) [inline kit/hydra/relay.py:89]: Doesn't this need to be on the same line?

File: kit/hydra/relay.py [hunk 2]
```diff
@@ -0,0 +1,305 @@
...
+]
+
+
+@lru_cache()
+def _camel_to_snake(name: str) -> str:
+    name = re.sub("(.)([A-Z][a-z]+)", r"\1_\2", name)
+    return re.sub("([a-z0-9])([A-Z])", r"\1_\2", name).lower()
+
+
+@dataclass(init=False)
```
[the remark this answers is not in the document]
Comment (MylesBartlett) [reply kit/hydra/relay.py:44]: Faster access/creation

File: kit/hydra/relay.py (hunk 1 above)
Comment (MylesBartlett) [reply kit/hydra/relay.py:89]: No, the block continues until the next empty line

File: kit/hydra/relay.py [hunk 3]
```diff
@@ -0,0 +1,305 @@
...
+
+from .utils import SchemaRegistration
+
+__all__ = [
+    "Relay",
+    "Option",
+]
+
+
+@lru_cache()
```
[the remark this answers is not in the document]
Comment (MylesBartlett) [reply kit/hydra/relay.py:38]: The default value for max_size is 128?

**Pull request merged**